uaclient/config.py

```diff
@@ -660,6 +661,34 @@ def parse_config(config_path=None):
     return cfg
 
 
+def apply_config_settings_override(override_key: str):
+    """Decorator used to override function return by config settings.
+
+    To identify if we should override the function return, we check
+    if the config object has the expected override key, we use it
+    has, we will use the key value as the function return. Otherwise
+    we will call the function normally.
+
+    @param override_key: key to be looked for in the settings_override
+     entry in the config dict. If that key is present, we will return
+     its value as the function return.
+    """
+
+    def wrapper(f):
+        @wraps(f)
+        def new_f():
+            cfg = parse_config()
+            value_override = cfg.get("settings_overrides", {}).get(
+                override_key
```

Comment:
@lucasmoura let's set a known unset sentinel value if override_key like 
UNSET = "_unset"                                                                

That way it would still allow someone to override "cloud_type : null" in uaclient.conf